src/solvers/lowering/byte_operators.cpp

```diff
@@ -478,7 +517,33 @@ exprt lower_byte_extract(const byte_extract_exprt &src, const namespacet &ns)
       return simplify_expr(vector, ns);
     }
   }
-  else if(ns.follow(src.type()).id()==ID_struct)
+  else if(src.type().id() == ID_complex)
+  {
+    const complex_typet &complex_type = to_complex_type(src.type());
+    const typet &subtype = complex_type.subtype();
+
+    auto subtype_bits = pointer_offset_bits(subtype, ns);
+    if(subtype_bits.has_value())
+    {
+      byte_extract_exprt real(unpacked);
```

Comment:
If `imag` is at (unpacked.offset + subtype_bytes), surely `real` is at `unpacked.offset`?

Reply:
(and if so, test needs to cover the non-zero-offset case)

Reply:
Yes, but that's already baked into `unpacked`, so no need to change the `offset`. I'll add a comment to clarify.